### PR TITLE
Add custom repo

### PR DESCRIPTION
**Pull request by lupintrd**
Hi!

This PR will add the possibility to set a custom gitlab-runner repository. It's useful if you have a local mirror and don't have direct internet access on the node you want to install it.
Additionally it will add a service resource to let systemd manage it.
Also it will fix a small issue with the name pattern in runner.pp .

Bye,
Johannes

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -27,6 +41,8 @@
   Boolean                    $manage_repo              = true,
   String                     $package_ensure           = installed,
   String                     $package_name             = 'gitlab-runner',
+  String                     $repo_base_url            = 'https://packages.gitlab.com',
+  String                     $repo_keyserver           = 'keys.gnupg.net',
```
Comment (bastelfreak) [inline manifests/init.pp]: you can probably use this datatype https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/fqdn.pp

File: manifests/init.pp [hunk 2]
```diff
@@ -27,6 +41,8 @@
   Boolean                    $manage_repo              = true,
   String                     $package_ensure           = installed,
   String                     $package_name             = 'gitlab-runner',
+  String                     $repo_base_url            = 'https://packages.gitlab.com',
```
Comment (bastelfreak) [inline manifests/init.pp]: you can probably use this type https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/httpsurl.pp

File: metadata.json [hunk 3]
```diff
@@ -15,6 +15,14 @@
     {
       "name": "puppetlabs/stdlib",
       "version_requirement": ">= 4.13.0 < 6.0.0"
+    },
+    {
+      "name": "puppetlabs/apt",
+      "version_requirement": ">= 6.3.0 < 7.0.0"
+    },
+    {
+      "name": "puppetlabs-docker",
```
Comment (bastelfreak) [inline metadata.json]: I think the docker module isn't always used in this module, so it's a soft dependency. Please don't add it to metadata.json.

File: metadata.json [hunk 4]
```diff
@@ -15,6 +15,14 @@
     {
       "name": "puppetlabs/stdlib",
       "version_requirement": ">= 4.13.0 < 6.0.0"
+    },
+    {
+      "name": "puppetlabs/apt",
+      "version_requirement": ">= 6.3.0 < 7.0.0"
```
Comment (bastelfreak) [inline metadata.json]: the newest release is 7.0.1. Is there a reason why you don't allow 7.x?

**Comment by bastelfreak** [comment]
Hey @lupintrd, thanks for the PR. Can you please take a look at the used email address in the commits? It isn't associated with your github account.

File: metadata.json (hunk 3 above)
Comment (lupintrd) [reply metadata.json]: Yes, you're right. I'll remove it righ away.

File: metadata.json (hunk 4 above)
Comment (lupintrd) [reply metadata.json]: Well, right. Minor updates should be ok. :)

File: manifests/init.pp (hunk 2 above)
Comment (lupintrd) [reply manifests/init.pp]: Yes, done.

File: manifests/init.pp (hunk 1 above)
Comment (lupintrd) [reply manifests/init.pp]: Yes, done for keyserver, Thanks.

File: metadata.json [hunk 5]
```diff
@@ -15,6 +15,10 @@
     {
       "name": "puppetlabs/stdlib",
       "version_requirement": ">= 4.13.0 < 6.0.0"
+    },
+    {
+      "name": "puppetlabs/apt",
+      "version_requirement": ">= 6.3.0"
```
Comment (bastelfreak) [inline metadata.json]: we need an upper version boundary here as well

Comment (lupintrd) [reply metadata.json]: Ok, may the Documentation be with me :)

**Comment by lupintrd** [comment]
Hi @bastelfreak, could you please check if this is good to go?
Thanks a lot,
Johannes

**Comment by lupintrd** [comment]
Hi @bastelfreak, i've merged back master and resolved an occurred conflict.
Regards,
Johannes

**Comment by bastelfreak** [comment]
thanks for the work!

**Pull request merged**